backend/src/openbeheer/types/_open_beheer.py

```diff
@@ -276,6 +278,44 @@ def __post_init__(self):
         self.name = camelize(self.name)
 
 
+def get_ob_subfields(
+    t: type | UnionType | Annotated,
+    option_overrides: Mapping[CamelCaseFieldName, list[OBOption]],
+    base_editable: Callable[[CamelCaseFieldName], bool],
+    meta: Meta | None = None,
+    prefix: str = "",
+) -> Iterable[OBField]:
+    args = get_args(t)
+    meta = meta or next((arg for arg in args if isinstance(arg, Meta)), None)
+
+    if args:
+        yield from get_ob_subfields(
+            next(ut for ut in args if ut not in (NoneType, UnsetType)),
+            option_overrides,
+            base_editable,
+            meta,
+            prefix,
+        )
+        return
+
+    assert isinstance(t, type)
+    if hasattr(t, "__struct_fields__"):
+        for field_name in t.__struct_fields__:
+            field = getattr(t, field_name)
+            prefixed_name: CamelCaseFieldName = camelize(prefix + field_name)
+
+            yield OBField(
+                name=prefixed_name,
+                type=as_ob_fieldtype(field, field_name),
+                options=option_overrides.get(prefixed_name, options(field)),
+                # TODO: we need to find a better way to handle this in the frontend before we enable editing :S
+                # editable=base_editable(prefixed_name)
+                # and not (_core_type(field) in READ_ONLY_TYPES),
+                editable=False,
+                required=_ob_required(t),
+            )
+
```

Comment:
This looks a lot like `ob_fields_of_type`.
No need to generalise it now, but a name that starts with `_` to show it shouldn't be used elsewhere would be nice.